game-core/src/main/java/games/strategy/util/LocalizeHtml.java

```diff
@@ -48,6 +50,7 @@ public static String localizeImgLinksInHtml(final String htmlText, final Resourc
     final Pattern patternLink = Pattern.compile(PATTERN_HTML_IMG_SRC_TAG);
     final Matcher matcherTag = patternTag.matcher(htmlText);
     Matcher matcherLink;
+    final List<String> alreadyReplaced = new ArrayList<>();
```

Comment:
I kind of expected to see a `Set` here.  I'm assuming you chose a list because the size of this collection is typically less than 10 elements?

I believe a `Set` would also let you merge the `contains()` and `add()` calls:

```java
if (!alreadyReplaced.add(link)) {
  break;
}
```